test/suite/test_prepare_discover04.py

```diff
@@ -0,0 +1,142 @@
+# ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+# OTHER DEALINGS IN THE SOFTWARE.
+#
+# test_prepare_discover04.py
+#   Test that prepare discover cursor can discover and commit a pending prepared delete
+
+import wiredtiger
+from suite_subprocess import suite_subprocess
+import wttest
+from wtscenario import make_scenarios
+
+@wttest.only_for_hook("disagg", "FIXME-WT-15343 disable RTS when precise checkpoint is on")
+class test_prepare_discover04(wttest.WiredTigerTestCase, suite_subprocess):
+    tablename = 'test_prepare_discover04'
+    uri = 'table:' + tablename
+    conn_config = 'precise_checkpoint=true,preserve_prepared=true'
+    s_config = 'key_format=i,value_format=S'
+
+    def test_prepare_discover_commit_delete(self):
+        self.conn.set_timestamp('stable_timestamp=' + self.timestamp_str(50))
+        self.conn.set_timestamp('oldest_timestamp=' + self.timestamp_str(50))
+        self.session.create(self.uri, self.s_config)
+        c = self.session.open_cursor(self.uri)
+
+        self.session.begin_transaction()
+        c[1] = "commit ts=60"
+        c[2] = "commit ts=60"
+        self.session.commit_transaction("commit_timestamp=" + self.timestamp_str(60))
+
+        session2 = self.conn.open_session()
+        c2 = session2.open_cursor(self.uri)
+        session2.begin_transaction()
+        c2.set_key(1)
+        self.assertEqual(c2.remove(), 0)
+        c2.set_key(2)
+        self.assertEqual(c2.remove(), 0)
+        session2.prepare_transaction('prepare_timestamp=' + self.timestamp_str(100) +',prepared_id=' + self.prepared_id_str(150))
+        # Move the stable timestamp to include the prepared transaction
+        self.conn.set_timestamp('stable_timestamp=' + self.timestamp_str(150))
+        # Create a checkpoint
+        session3 = self.conn.open_session()
+        session3.checkpoint()
+
+        # Creating backup that will preserve artifacts
+        backup_dir = 'bkp'
+        self.backup(backup_dir, session3)
+
+        # Opening backup database
+        conn2 = self.wiredtiger_open(backup_dir, self.conn_config)
+        c2s1 = conn2.open_session()
+
+        # Opening prepared discover cursor
+        prepared_discover_cursor = c2s1.open_cursor("prepared_discover:")
+
+        # Walking through prepared discover cursor
+        c2s2 = conn2.open_session()
+        count = 0
+        while prepared_discover_cursor.next() == 0:
+            count += 1
+            prepared_id = prepared_discover_cursor.get_key()
+            c2s2.begin_transaction("claim_prepared_id=" + self.timestamp_str(prepared_id))
+            c2s2.breakpoint()
+            c2s2.commit_transaction("commit_timestamp=" + self.timestamp_str(200)+",durable_timestamp=" + self.timestamp_str(210))
+
+        self.assertEqual(count, 1)
+        conn2.set_timestamp('stable_timestamp=' + self.timestamp_str(220))
+        c2s2.checkpoint()
+
+    def test_prepare_discover_rollback_delete(self):
```

Comment:
The test has no difference other than commit/rollback. How about adding that case as a scenario and, based on the flag, either doing the commit or rollback to avoid duplicate code?